public/manageclaims.php

```diff
@@ -0,0 +1,269 @@
+<?php
+require_once __DIR__ . '/../vendor/autoload.php';
+require_once __DIR__ . '/../lib/bootstrap.php';
+
+use RA\ArticleType;
+use RA\ClaimSetType;
+use RA\ClaimSorting;
+use RA\ClaimSpecial;
+use RA\ClaimStatus;
+use RA\ClaimType;
+use RA\Permissions;
+
+if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::Admin)) {
+    header("Location: " . getenv('APP_URL'));
+    exit;
+}
+
+$gameID = requestInputSanitized('g', null, 'integer');
+
+if (empty($gameID)) {
+    header("Location: " . getenv('APP_URL'));
+}
+
+$claimData = getFilteredClaimData($gameID);
+$gameData = getGameData($gameID);
+
+$consoleName = $gameData['ConsoleName'];
+$gameTitle = $gameData['Title'];
+$gameIcon = $gameData['ImageIcon'];
+
+RenderHtmlStart();
+RenderHtmlHead("Manage Claims");
+?>
+<body>
+<?php
+RenderHeader($userDetails);
+?>
+<link rel="stylesheet" href="/vendor/jquery.datetimepicker.min.css">
+<script src="/vendor/jquery.datetimepicker.full.min.js"></script>
+<script>
+
+  /**
+   * Creates update post message when a claim is updated by an admin
+   */
+  function UpdateClaimDetails(claimID, claimUser, claimType, setType, claimStatus, claimSpecial, claimDate, doneDate) {
+    var somethingChanged = 0;
+    var comment = "<?= $user ?> updated " + claimUser + "'s claim. ";
+
+    var newClaimType = parseInt($('#claimType_' + claimID).val());
+    if (newClaimType != claimType) {
+        comment += "Claim Type: " + (newClaimType == <?= ClaimType::Primary ?> ? "<?= ClaimType::toString(ClaimType::Primary) ?>. " : "<?= ClaimType::toString(ClaimType::Collaboration) ?>. ")
+        somethingChanged = 1;
+    }
+
+    var newSetType = parseInt($('#setType_' + claimID).val());
+    if (newSetType != setType) {
+        comment += "Set Type: " + (newSetType == <?= ClaimSetType::NewSet ?> ? "<?= ClaimSetType::toString(ClaimSetType::NewSet) ?>. " : "<?= ClaimSetType::toString(ClaimSetType::Revision) ?>. ")
+        somethingChanged = 1;
+    }
+
+    var newClaimStatus = parseInt($('#status_' + claimID).val());
+    if (newClaimStatus != claimStatus) {
+        switch (newClaimStatus) {
+            case <?= ClaimStatus::Active ?>:
+                comment += "Claim Status: <?= ClaimStatus::toString(ClaimStatus::Active) ?>. ";
+                break;
+            case <?= ClaimStatus::Complete ?>:
+                comment += "Claim Status: <?= ClaimStatus::toString(ClaimStatus::Complete) ?>. ";
+                break;
+            case <?= ClaimStatus::Dropped ?>:
+                comment += "Claim Status: <?= ClaimStatus::toString(ClaimStatus::Dropped) ?>. ";
+                break;
+            default:
+                comment += "Claim Status: <?= ClaimStatus::toString(ClaimStatus::Active) ?>. ";
+                break;
+        }
```

Comment:
I don't think I can do that here because of the js/php mixture. I don't know `newClaimStatus` until the update button is pressed and at that point the php code has already been processed.

Reply:
Completely missed that this was in javascript.